src/adapters/refract-adapter.coffee

```diff
@@ -50,7 +59,28 @@ transformAst = (element, sourcemap, options) ->
   applicationAst
 
 
+transformError = (source, parseResult) ->
+  errors = _.chain(parseResult)
+    .filterContent({element: 'annotation'})
+    .filter({meta: {classes: ['error']}})
+    .value()
+
+  if errors.length > 0
+    sourceMap = errors[0]?.attributes?.sourceMap?[0]?.content?[0]
+    index = sourceMap?[0]
+    length = sourceMap?[1]
+
+    error = {
+      message: errors?[0]?.content
+      code: errors?[0]?.attributes?.code or 1
+      line: countLines(source, index)
+      location: [{index, length}]
```

Comment:
Why is the location taking the first source map only? It should have all the sourcemaps.